File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        ones = s.count("1")
        zeroes = 0
        output = 0

        for i in range(len(s) - 1):
            char = s[i]
            if (char == "0"):
                zeroes += 1
            else:
                ones -= 1

            output = max(output, ones + zeroes)

        return output

File: 1537-maximum-score-after-splitting-a-string/maximum-score-after-splitting-a-string.js
var maxScore = function(s) {
    let zeroes = 0
    let ones = 0
    let output = 0

    for (let char of s) {
        if (char == "1") ones += 1
    }

    for (let i = 0; i < s.length - 1; i++) {
        let char = s[i]
        if (char == "1") ones -= 1
        else zeroes += 1
        score = zeroes + ones
        output = Math.max(output, score) 
    }

    return output
};